server-common/src/main/java/org/apache/kafka/server/common/serialization/AbstractApiMessageSerde.java

```diff
@@ -69,16 +76,42 @@ public void write(ApiMessageAndVersion data,
     @Override
     public ApiMessageAndVersion read(Readable input,
                                      int size) {
-        short frameVersion = (short) input.readUnsignedVarint();
+        short frameVersion;
+        try {
+            frameVersion = unsignedIntToShort(input.readUnsignedVarint(), "frame version");
+        } catch (Exception e) {
+            throw new MetadataParseException("Failed to read variable-length " +
```

Comment:
It looks like `MetadataParseException` is already the only thing which can be raised by `unsignedIntToShort`. Is it necessary to catch it and raise a new instance?

Reply:
In fact `input.readUnsignedVarint()` may throw a IlleagelArgumentException if it fails to read a short, this can be seen in `MetadataRecordSerdeTest.testParsingMalformedFrameVersionVarint`.